meson.build

```diff
@@ -34,6 +35,12 @@ _global_c_args = cc.get_supported_arguments(
 )
 add_project_arguments(_global_c_args, language : 'c')
 
+limited_api = ''
+if get_option('python.allow_limited_api')
+  add_project_arguments('-DPy_LIMITED_API=0x030C0000', language : ['c', 'cpp'])
```

Comment:
This isn't actually necessary I found; Meson will add this define to the extension module targets when you give it `limited_api = '3.12'`.